### PR TITLE
#165287338 A user will be able to edit their profile

### PR DESCRIPTION
**Pull request by toluola**
#### Description
- A user will be able to edit their Profile  

#### Type of change
-  New feature (a non-breaking change which adds functionality)

#### How Has This Been Tested?
- Integration

#### Checklist:
- [x] my code follows style guidelines for this project

#### PT
- [#165287338](https://www.pivotaltracker.com/story/show/165287338)

#### Postman Snapshot
Screenshot 2019-04-13 at 8 31 53 PM

### CODE REVIEW

File: src/routers/index.js [hunk 1]
```diff
@@ -57,7 +59,9 @@ router.get('/auth/google/callback', passport.authenticate('google', { failureRed
  */
 
 router
```
Comment (chikeozulumba) [inline src/routers/index.js]: @toluola Nice job, you didn't add a function description

File: src/controllers/authentication/user.js [hunk 2]
```diff
@@ -42,18 +42,53 @@ export const createUser = async (req, res) => {
...
     return res.status(500).json(errorResponseFormat({
+      status: 'error',
+      message: 'Something Went Wrong',
+    }));
+  }
+};
+
+export const editUser = async (req, res) => {
+  try {
+    const { fullName, bio, imageUrl, notification, username } = req.body;
```
Comment (ssanusi) [inline src/controllers/authentication/user.js]: @toluola  it is better we get users details from the token other than request body that way we are sure the users are editing profile that they used to login

**Review by ssanusi — COMMENTED** [review]
@toluola this ticket says user should be able to edit there profile and from your implementation i think editing just a single profile field is not considered

File: src/controllers/authentication/user.js (hunk 2 above)
Comment (toluola) [reply src/controllers/authentication/user.js]: @ssanusi we can not get the details that the user wants to edit from their token, we can only get their ID from the token to be sure the user is the one doing the editing and not someone else.

File: src/controllers/authentication/user.js [hunk 3]
```diff
@@ -43,18 +43,21 @@ export const createUser = async (req, res) => {
...
       if (error.fields.username) {
         return res.status(409).json(errorResponseFormat({
+          status: 'error',
           message: 'This Username Already Exist',
         }));
       }
     }
 
     return res.status(500).json(errorResponseFormat({
+      status: 'error',
```
Comment (Temmyogunbo) [inline src/controllers/authentication/user.js]: Is this a new convention?

**Review by Temmyogunbo — DISMISSED** [review]

**Review by Temmyogunbo — APPROVED** [review]

**Pull request merged**